testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -1261,6 +1261,53 @@ public void testInsertDeletUpdateWithPrestoAndHive()
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = 60 * 60 * 1000, dataProvider = "transactionalNonTransactionalProvider")
+    public void testAnalyzeNonPartitionedTable(boolean transactional)
+    {
+        withTemporaryTable("test_analyze", true, true, NONE, targetTable -> {
+            onPresto().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchases INT, address VARCHAR) WITH (transactional = %s)", targetTable, transactional));
+
+            log.info("Inserting a few rows");
+            onPresto().executeQuery(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 5, 'Antioch'), ('Bill', 7, 'Buena'), ('Carol', 3, 'Cambridge'), ('Dave', 11, 'Devon')", targetTable));
+
+            log.info("Inserting a few more");
+            onPresto().executeQuery(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 6, 'Arches'), ('Carol', 9, 'Centreville'), ('Dave', 11, 'Darbyshire'), ('Ed', 7, 'Etherville')", targetTable));
+
+            log.info("Running ANALYZE TABLE");
+            onPresto().executeQuery("ANALYZE " + targetTable);
+
+            log.info("Finished");
+        });
+    }
+
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = 60 * 60 * 1000, dataProvider = "transactionalNonTransactionalProvider")
+    public void testAnalyzePartitionedTable(boolean transactional)
+    {
+        withTemporaryTable("test_analyze", true, true, NONE, targetTable -> {
+            onPresto().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchases INT, address VARCHAR) WITH (transactional = %s, partitioned_by = ARRAY['address'])", targetTable, transactional));
+
+            log.info("Inserting a few rows");
+            onPresto().executeQuery(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 5, 'Antioch'), ('Bill', 7, 'Buena'), ('Carol', 3, 'Cambridge'), ('Dave', 11, 'Devon')", targetTable));
+
+            log.info("Inserting a few more");
+            onPresto().executeQuery(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 6, 'Arches'), ('Carol', 9, 'Centreville'), ('Dave', 11, 'Darbyshire'), ('Ed', 7, 'Etherville')", targetTable));
+
+            log.info("Running ANALYZE TABLE");
+            onPresto().executeQuery("ANALYZE " + targetTable);
+
+            log.info("Finished");
+        });
+    }
+
+    @DataProvider
+    public Object[][] transactionalNonTransactionalProvider()
+    {
+        return new Object[][] {
+                {false},
```

Comment:
ANALYZE for non-transactional tables is already covered in `io.trino.tests.hive.TestHiveBasicTableStatistics#testAnalyzePartitioned` and `io.trino.tests.hive.TestHiveBasicTableStatistics#testAnalyzePartitioned`.

I think it could be useful to consolidate the testing there, by adding `boolean transactional` param, dataprovider, and something like
```
if (transactional) {
    ensureTransactionalHive();
}
```

but i will let @losipiuk @wendigo share their opinion too

Reply:
I haven't yet addressed this comment.